src/coreclr/hosts/coreshim/ComActivation.cpp

```diff
@@ -60,22 +59,29 @@ STDAPI DllGetClassObject(
         "GetClassFactoryForTypeInternal", (void**)&GetClassFactoryForTypeInternal));
 
     // Get assembly and type for activation
-    std::string assemblyName;
+    std::wstring assemblyName;
     RETURN_IF_FAILED(Utility::TryGetEnvVar(COMACT_ASSEMBLYNAME_ENVVAR, assemblyName));
 
-    std::string typeName;
+    std::wstring typeName;
     RETURN_IF_FAILED(Utility::TryGetEnvVar(COMACT_TYPENAME_ENVVAR, typeName));
 
+    // Compute the path to the assembly. This should be adjacent to CoreShim (i.e. this library).
+    std::wstring assemblyPath;
+    RETURN_IF_FAILED(Utility::GetCoreShimDirectory(assemblyPath));
+    assemblyPath.append(assemblyName);
+    assemblyPath.append(W(".dll"));
+
     IUnknown *ccw = nullptr;
 
     struct ComActivationContext
     {
         GUID ClassId;
         GUID InterfaceId;
+        const void *AssemblyPath;
```

Comment:
Yeah I can. I was trying to avoid relying on auto marshalling for these things, but I can do that.

Reply:
Well this declaration is in C++, so no auto-marshalling applies.. right?
In the managed side I'm fine with the manual approach.

Reply:
> In the managed side I'm fine with the manual approach.

Oh I thought you were advocating to change both sides. I can do that.

>... and on the managed side, it can be char* to make it easier to understand.

Never thought of using an unsafe type on the managed side. I can do that.

Reply:
`IntPtr` is as unsafe as `char*` in interop. 

The fact that C# makes you write `unsafe` for one, but not the other, is misleading. It is unfortunate that C# `unsafe` is limited to unsafe C# language, and does not capture library calls, etc.

Reply:
Yep, that is what I meant by "unsafe type". I have historically been averse to using the `unsafe` keyword purely because it requires changing compiler flags. In this case it doesn't matter, but that is why I tend to not consider using `unsafe` at all.